source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -359,44 +359,48 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
         return S_OK;
     }
 
-    void ApplyBackgroundToRoot(_In_ IPanel* rootPanel,
-                               _In_ IAdaptiveBackgroundImage* backgroundImage,
-                               _In_ IAdaptiveRenderContext* renderContext,
-                               _In_ IAdaptiveRenderArgs* renderArgs)
+    void CreateBackgroundImage(_In_ IAdaptiveRenderContext* renderContext, _In_ HSTRING url, _Outptr_ IImage** backgroundImage)
     {
-        // In order to reuse the image creation code paths, we simply create an adaptive card
-        // image element and then build that into xaml and apply to the root.
-        ComPtr<IAdaptiveImage> adaptiveImage = XamlHelpers::CreateABIClass<IAdaptiveImage>(
-            HStringReference(RuntimeClass_AdaptiveCards_ObjectModel_Uwp_AdaptiveImage));
+        ComPtr<IImage> image = XamlHelpers::CreateABIClass<IImage>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Image));
 
-        HString url;
-        THROW_IF_FAILED(backgroundImage->get_Url(url.GetAddressOf()));
-        THROW_IF_FAILED(adaptiveImage->put_Url(url.Get()));
+        ComPtr<IDependencyObject> imageAsDependencyObject;
+        THROW_IF_FAILED(image.As(&imageAsDependencyObject));
 
-        ComPtr<IAdaptiveCardElement> adaptiveCardElement;
-        THROW_IF_FAILED(adaptiveImage.As(&adaptiveCardElement));
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        THROW_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
 
-        ComPtr<IAdaptiveElementRendererRegistration> elementRenderers;
-        THROW_IF_FAILED(renderContext->get_ElementRenderers(&elementRenderers));
+        ComPtr<IUriRuntimeClass> imageUrl;
+        GetUrlFromString(hostConfig.Get(), url, imageUrl.GetAddressOf());
```

Comment:
@paulcam206 should I do this in **applyBackgroundToRoot** and pass an actual working url here?